res_partner_statement/wizard/partner_statement_wizard.py

```diff
@@ -170,18 +170,25 @@ def _get_lines_print(self, partner):
             'sales_total': 0.0,
             'balance': 0.0
         }
+        lines_to_print = []
         for invoice in invoices:
-            if invoice.type in ['out_refund', 'in_refund']:
-                items, totals = self.items(
-                    invoice, totals, _('Credit Note'))
-            else:
-                items, totals = self.items(
-                    invoice, totals, _('Invoice'))
+            if invoice not in [x[0] for x in lines_to_print]:
+                if invoice.type in ['out_refund', 'in_refund']:
+                    lines_to_print.append([invoice, _('Credit Note')])
+                else:
+                    lines_to_print.append([invoice, _('Invoice')])
+                for payment in invoice.payment_move_line_ids:
+                    if payment not in [y[0] for y in lines_to_print]:
+                        lines_to_print.append([payment, _('Payment')])
+                    else:
+                        for line in lines_to_print:
+                            if payment == line[0]:
```

Comment:
too many nested blocks in travis.
You need to refactor an create another function.